nextcloudappstore/core/static/assets/js/user-api-token.js

```diff
@@ -0,0 +1,100 @@
+(function (global) {
+    'use strict';
+
+    function regenAuthToken(url, api_token) {
+        let request = new Request(
+            url,
+            {
+                method: 'POST',
+                headers: new Headers({
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Token ' + api_token
+                }),
+            }
+        );
+        return fetch(request).then(global.convertResponse);
+    };
+
+
+    function fetchToken(csrf) {
+        return global.fetchAPIToken(csrf).then(
+            (response) => response.token,
+            () => null
```

Comment:
why not return the promise itself without the do nothing error handler? than the caller can add a catch() method to handle the error